libvirt/cloudinit_def.go

```diff
@@ -62,7 +62,11 @@ func (ci *defCloudInit) CreateAndUpload(virConn *libvirt.Connect) (string, error
 	}
 	defer pool.Free()
 
-	PoolSync.AcquireLock(ci.PoolName)
+	for {
+		if PoolSync.AcquireLock(ci.PoolName) {
```

Comment:
Don't know how to avoid looping in this case. I added a `time.Sleep(..)` to make it more CPU-friendly ;)